### PR TITLE
Tag improvements

### PR DESCRIPTION
**Pull request by varnastadeus**
Solves problems in #146 
1. Not able to add tag if there are items matching search term
2. Selected items are visible among options even if they don't match search term

### CODE REVIEW

File: src/ng-select/ng-select.component.ts [hunk 1]
```diff
@@ -621,11 +626,27 @@ export class NgSelectComponent implements OnInit, OnDestroy, OnChanges, AfterVie
...
+            this.itemsList.unmarkItem();
+            this.dropdownList.scrollIntoTag();
+        } else {
+            this.itemsList.markPreviousItem();
+            this.scrollToMarked();
+        }
         $event.preventDefault();
     }
 
+    private nextItemIsTag(nextStep: number) {
```
Comment (anjmao) [inline src/ng-select/ng-select.component.ts]: This is not question but statement. Maybe better `isTagItem`?

File: src/ng-select/virtual-scroll.component.ts [hunk 2]
```diff
@@ -163,6 +163,13 @@ export class VirtualScrollComponent implements OnInit, OnChanges, OnDestroy {
         this.refresh();
     }
 
+    scrollIntoTag() {
+        let el: Element = this.element.nativeElement;
```
Comment (anjmao) [inline src/ng-select/virtual-scroll.component.ts]: Why `el` is not const?

**Comment by anjmao** [comment]
Tested. Not found any bug. Good job!

**Comment by varnastadeus** [comment]
@anjmao `Clear items on open if is typeahead mode` we can't do that since we can use typeahead to search in multiple fields without async data

**Pull request merged**